Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Location/LocationDrawingsTable.tsx

```diff
@@ -0,0 +1,188 @@
+        return true;
+    }
+
+    const fetchDrawings = (sortKey: keyof SystemCenter.Types.LocationDrawing, ascending: boolean, page: number, locationID: number) => {
+        setPageState('loading');
+        const handle = LocationDrawingController.PagedSearch([], sortKey, ascending, page, locationID)
+            .done((result) => {
+                setLinks(JSON.parse(result.Data as unknown as string));
+                if (result.NumberOfPages === 0) result.NumberOfPages = 1;
+                setPageInfo({RecordsPerPage: result.RecordsPerPage, 
+                            NumberOfPages: result.NumberOfPages,
+                            TotalRecords: result.TotalRecords});
+                setPageState('idle');
+            })
+            .fail(() => setPageState('error'));
+        return handle;
+    }
+
+    const handleDelete = (item: SystemCenter.Types.LocationDrawing) => {
+        setPageState('loading');
+        LocationDrawingController.DBAction('DELETE', item)
+            .then(() => fetchDrawings(sortKey, ascending, page, props.LocationID),
+                  () => setPageState('error'))
+    };
+
+    React.useEffect(() => {
+        const storedInfo = JSON.parse(localStorage.getItem(PagingID) as string);
+        if (storedInfo != null) setPage(storedInfo);
+    }, []);
+
+    React.useEffect(() => {
+        localStorage.setItem(PagingID, JSON.stringify(page));
+    }, [page]);
+
+    React.useEffect(() => {
+        const handle = fetchDrawings(sortKey, ascending, page, props.LocationID);
+        return () => { if (handle != null && handle?.abort != null) handle.abort(); }
+    }, [sortKey, ascending, page, props.LocationID, props.RefreshDrawings]);
+
+    return <>
+        <div className="w-100 h-100 d-flex flex-column">
+            <ReactTable.Table<SystemCenter.Types.LocationDrawing>
+                TableClass="table table-hover"
+                Data={links}
+                SortKey={sortKey}
+                Ascending={ascending}
+                OnSort={(d) => {
+                    if (d.colKey === 'EditDelete')
+                        return;
+                    if (d.colKey == sortKey)
+                        setAscending(!ascending);
+                    else {
+                        setAscending(true);
+                        setSortKey(d.colKey as keyof SystemCenter.Types.LocationDrawing);
+                    }
+                }}
+                TableStyle={{ padding: 0, width: '100%', tableLayout: 'fixed', display: 'flex', flexDirection: 'column', overflow: 'hidden', flex: 1 }}
+                TheadStyle={{ fontSize: 'smaller', display: 'table', tableLayout: 'fixed', width: '100%' }}
+                TbodyStyle={{ display: 'block', width: '100%', overflowY: 'auto', flex: 1 }}
+                RowStyle={{ fontSize: 'smaller', display: 'table', tableLayout: 'fixed', width: '100%' }}
+                Selected={(item) => false}
+                KeySelector={(item) => item.ID}
+            >
```

Comment:
Some columns here are very wide comparied to info they show and some are very narrow. Can we switch some of these from auto to pixels or percentages?

Reply:
updated in latest commit